### PR TITLE
Firmware fixes - iwlwifi, b43* and ipw

### PR DESCRIPTION
**Pull request by 01micko**
- fix iwlwifi - look harder for a compatible version
- change 'non-free' to 'missing' in `kernel-kit/firmware_picker.sh`
- b43[legacy] - add to `kernel-kit/output/fdrv*` (optional - see `build.conf`)
- ipw (Intel Pro Wireless) - ditto
- remove 'non-free' firmware code from `3builddistro`, `support/` and configs

### CODE REVIEW

File: kernel-kit/firmware_extra.sh [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	if [ "$save_dld" = "true" ];then
+		[ -e "${REPO}/${PKG}" ] && cp -af ${REPO}/${PKG} ${TEMP} && return $? # bale here
+		TGT=${REPO}
+	fi
+	[ -z "$3" ] && OPT="-P ${TGT}" || OPT="-O ${TGT}/${PKG}"
+	wget ${PKGURL} ${OPT} || return 1
+	[ ! -e "${TEMP}/${PKG}" ] && cp -af ${REPO}/${PKG} ${TEMP} || return 1
+}
+
+B43_URL='http://www.lwfinger.com/b43-firmware/broadcom-wl-6.30.163.46.tar.bz2'
```
Comment (dimkr) [inline kernel-kit/firmware_extra.sh]: HTTPS?

File: kernel-kit/firmware_picker.sh [hunk 2]
```diff
@@ -73,41 +73,75 @@ cd ${CWD}
...
-			fi
+			case $fw in
+				iwlwifi*) # some iwlwifi versions differ from modinfo
+				fw_ver=${fw%\.*}
+				fw_ver=${fw_ver##*\-}
+				case $fw_ver in
+					''|*[!0-9]*)continue ;; # skip as version is not an integer
+					*)
+					c=$fw_ver
+					limit=$(($c - 12))
```
Comment (dimkr) [inline kernel-kit/firmware_picker.sh]: Why 12?

Comment (01micko) [reply kernel-kit/firmware_picker.sh]: I thought there was another about 10 versions away. Might reduce to 4.

File: kernel-kit/firmware_extra.sh (hunk 1 above)
Comment (01micko) [reply kernel-kit/firmware_extra.sh]: Yes, should be. Will do.

**Review by dimkr — APPROVED** [review]

**Comment by 01micko** [comment]
I want to merge this but can't find a relevant broadcom licence!

These firmwares are over 10 years old so I think I'll merge anyway and deal with it later.. @dimkr ? What do you think?

**Comment by dimkr** [comment]
Take a look at the Debian package, they always have a license inside.

**Comment by dimkr** [comment]
https://gitweb.gentoo.org/repo/gentoo.git/plain/licenses/Broadcom

**Comment by 01micko** [comment]
Thanks, I'll sort it this weekend

**Pull request merged**